src/test/java/org/isf/menu/rest/UserControllerTest.java

```diff
@@ -0,0 +1,845 @@
+			groupPermission.setPermission(permission);
+
+			when(userManager.findUserGroupByCode(any())).thenReturn(userGroup);
+			when(permissionManager.retrievePermissionById(anyInt())).thenReturn(permission);
+			when(groupPermissionManager.create(any(), any())).thenReturn(groupPermission);
+
+			var result = mvc.perform(
+							post("/users/groups/{group_code}/permissions/{id}", userGroup.getCode(), permission.getId())
+									.contentType(MediaType.APPLICATION_JSON)
+					)
+					.andDo(log())
+					.andExpect(status().isCreated())
+					.andExpect(content().string(containsString("true")))
+					.andReturn();
+
+			LOGGER.debug("result: {}", result);
+		}
+
+		@Test
+		@WithMockUser(username = "admin", authorities = { "usergroups.create", "grouppermission.create" })
+		@DisplayName("Assign already assigned permission to a user group")
+		void assignAlreadyAssignedPermissionToUserGroup() throws Exception {
+			UserGroup userGroup = new TestUserGroup().setup(false);
+			Permission permission = TestPermission.generatePermission();
+			GroupPermission groupPermission = new GroupPermission();
+			groupPermission.setUserGroup(userGroup);
+			groupPermission.setPermission(permission);
+
+			when(userManager.findUserGroupByCode(any())).thenReturn(userGroup);
+			when(permissionManager.retrievePermissionById(anyInt())).thenReturn(permission);
+			when(groupPermissionManager.create(any(), any())).thenThrow(
+					new OHDataValidationException(new OHExceptionMessage(""))
+			);
+
+			mvc.perform(
+					post("/users/groups/{group_code}/permissions/{id}", userGroup.getCode(), permission.getId())
+							.contentType(MediaType.APPLICATION_JSON)
+			)
+			.andDo(log())
+			.andExpect(status().is4xxClientError())
+			.andExpect(content().string(containsString("Failed")));
+		}
+
+		@Test
+		@WithMockUser(username = "admin", authorities = { "usergroups.create", "grouppermission.create" })
+		@DisplayName("Assign non existing permission to a user group")
+		void assignNonExistingPermissionToUserGroup() throws Exception {
+			UserGroup userGroup = new TestUserGroup().setup(false);
+			Permission permission = TestPermission.generatePermission();
+			GroupPermission groupPermission = new GroupPermission();
+			groupPermission.setUserGroup(userGroup);
+			groupPermission.setPermission(permission);
+
+			when(userManager.findUserGroupByCode(any())).thenReturn(null);
+			when(permissionManager.retrievePermissionById(anyInt())).thenReturn(null);
+			when(groupPermissionManager.create(any(), any())).thenReturn(groupPermission);
+
+			mvc.perform(
+							post("/users/groups/{group_code}/permissions/{id}", userGroup.getCode(), permission.getId())
+									.contentType(MediaType.APPLICATION_JSON)
+					)
+					.andDo(log())
+					.andExpect(status().is4xxClientError())
+					.andExpect(content().string(containsString("not found")));
+		}
+
+		@Test
+		@WithMockUser(username = "admin", authorities = { "usergroups.create", "grouppermission.create" })
```

Comment:
Should this be like?

```suggestion
		@WithMockUser(username = "admin", authorities = { "grouppermission.delete" })
```

Reply:
Hi @SilverD3, what about this?